Monetization (MON-367): MonetizationModule - added MonetizationService to AdEngine

saipetch opened the pull request:
Monetization (MON-367): add MonetizationService to AdEngine and update unit tests
(replace #7045)

Commit:
Monetization (MON-367): MonetizationModule - added MonetizationService to AdEngine

Commit:
Monetization (MON-367): AdEngine - updated code to use injectHtml() function and added unit test

Commit:
Monetization (MON-367): AdEngine - added unit tests for MonetizationService

Commit:
Monetization (MON-367): AdEngine - removed wgUsePostScribe for MonetizationService

Commit:
Merge branch 'dev' into MON-367

Commit:
Merge branch 'dev' into MON-367

Commit:
Merge branch 'dev' into MON-367

nandy-andy commented on extensions/wikia/AdEngine/AdEngine2ContextService.class.php:
```diff
@@ -24,6 +24,11 @@ public function getContext( Title $title, $skinName ) {
 
+			$monetizationServiceAds = null;
+			if ( !empty( $wg->AdDriverUseMonetizationService ) && !empty( $wg->EnableMonetizationModuleExt ) ) {
+				$monetizationServiceAds = F::app()->sendRequest( 'MonetizationModule', 'index' )->getData()['data'];
```
I'm going to nitpick a little bit. I hope you'll forgive me :japanese_ogre: 

What happens when there is error or empty response from Monetization service?

jimmordino replied:
[Errors](http://graph-s3/grafana/#/dashboard/db/Monetization%20Service?panelId=5&fullscreen&from=now-24h&to=now)?  :smiling_imp:

nandy-andy commented on extensions/wikia/AdEngine/js/config/desktop.js:
```diff
@@ -12,6 +12,7 @@ define('ext.wikia.adEngine.config.desktop', [
 	'ext.wikia.adEngine.provider.evolve',
 	'ext.wikia.adEngine.provider.liftium',
```
Note to Ad Engineering: liftium is placed wrongly. Next time we touch `config/desktop.js` we could move it down so we keep alphabetical order.

nandy-andy commented on extensions/wikia/AdEngine/js/provider/monetizationService.js:
```diff
@@ -0,0 +1,54 @@
+		if (context.providers.monetizationServiceAds && context.providers.monetizationServiceAds[slotName]) {
+			log(['fillInSlot', slot, 'injectScript'], 'info', logGroup);
+
+			monetizationService.loadAssets();
```
If I understand correctly the assets are cached so the request for them is being sent only once: for the first slot. What do you think about placing a comment about it above?

nandy-andy commented on skins/oasis/modules/templates/Body_Index.php:
```diff
@@ -139,9 +139,9 @@
-						echo $monetizationModules[MonetizationModuleHelper::SLOT_TYPE_BELOW_CATEGORY];
-					}
+				if ( !empty( $wg->EnableMonetizationModuleExt ) && !empty( $monetizationModules[MonetizationModuleHelper::SLOT_TYPE_BELOW_CATEGORY] ) ) {
+					echo $monetizationModules[MonetizationModuleHelper::SLOT_TYPE_BELOW_CATEGORY];
```
Why don't we have here the same logic as below for `MON_ABOVE_FOOTER`?

saipetch replied:
Because below category slot uses different rule.

Commit:
MON-367 Small CR fixes

Removed localStorage-based caching from loadAssets (the assets are
cached on HTTP level, no need for duplication).

Removed trailing commas in define line.

Moved loadAsset to monetization service provider and removed
MonetizationServiceHelper.js

gabrys commented:
@saipetch I updated the pull request with a few tweaks, see the commit description

gabrys commented:
@saipetch: is there any reason for this PR being against MonetizationModule, not dev?

And one more question: the loadAssets is asynchronous and will be loaded in parallel to the actual Amazon/AdSense creatives. Is there a need to synchronize those, so that the ads load after the CSS and JS are loaded and executed (or the other way round)?
